lib/daru/index.rb

```diff
@@ -23,19 +23,14 @@ class << subclass
     def self.new *args, &block
       source = args[0]
 
-      idx =
-        if source && source[0].is_a?(Array)
-          Daru::MultiIndex.from_tuples source
-        elsif source && source.is_a?(Array) && !source.empty? &&
-              source.all? { |e| e.is_a?(DateTime) }
-          Daru::DateTimeIndex.new(source, freq: :infer)
-        else
-          i = allocate
-          i.send :initialize, *args, &block
-          i
-        end
-
-      idx
+      if source && source[0].is_a?(Array)
```

Comment:
Nope, we should never. 
`&&` is logical operator, `and` is control flow operator (which, in fact, Rubocop bans even for control flow, but I'm not completely agree about it).